File: backend/src/main/java/backend/capstone/domain/mobility/dayroute/dto/UploadTrigger.java
package backend.capstone.domain.mobility.dayroute.dto;

import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;

// 안드로이드가 GPS 배치 업로드를 어떤 이유로 트리거했는지 나타내는 관측용 값.
// 클라이언트가 아직 서버가 모르는 새 트리거 값을 보내더라도 업로드 자체는 실패하지 않도록,
// 인식 못한 값은 UNKNOWN으로 받는다(READ_UNKNOWN_ENUM_VALUES_USING_DEFAULT_VALUE 필요).
public enum UploadTrigger {
    WATCHING_IMMEDIATE,
    BATCH_SIZE,
    PERIODIC,
    PRE_BOUNDARY,
    NETWORK_RECOVERY,
    STOP_FLUSH,
    @JsonEnumDefaultValue
    UNKNOWN
}

File: android/app/src/main/java/com/example/passedpath/feature/locationtracking/domain/policy/UploadTrigger.kt
package com.example.passedpath.feature.locationtracking.domain.policy

// GPS 배치 업로드가 어떤 이유로 트리거됐는지 나타내는 관측용 사유.
// 서버로 그대로 전달되어 디버깅/관측 로그에 남는다.
enum class UploadTrigger {
    WATCHING_IMMEDIATE, // 보호자가 보고 있어(watching) 즉시 업로드 모드일 때
    BATCH_SIZE,         // 대기 좌표가 배치 크기에 도달해 즉시 업로드될 때
    PERIODIC,           // 3분 주기 루프
    PRE_BOUNDARY,       // 날짜 경계 직전 flush
    NETWORK_RECOVERY,   // 네트워크 복구 시 flush
    STOP_FLUSH          // 서비스 종료 시 flush
}
